SLING-12004 - ServiceUserMapper fails to build with Java 17

rombert opened the pull request:
- switch to a recent version of mockito
- update to a recent enough version of the parent pom

kwin commented on pom.xml:
```diff
@@ -23,7 +23,7 @@
         <groupId>org.apache.sling</groupId>
         <artifactId>sling-bundle-parent</artifactId>
-        <version>41</version>
+        <version>49</version>
```
why not most recent version 52?

rombert replied:
Because after a couple of minutes I could not figure out which individual artifacts to replace osgi.core 7.0.0 with. Do you have a suggestion?

kwin replied:
Usually https://docs.osgi.org/artifacts/#individual-artifacts provides enough information. Just start with framework (https://github.com/apache/sling-parent/blob/b8a8123e95e59762372db37cc1dd2ce15645f7a5/sling-bundle-parent/pom.xml#L242) and then add the following ones from the parent dependency management if needed.

rombert replied:
That works, thanks. See [b709ec3](https://github.com/apache/sling-org-apache-sling-serviceusermapper/pull/9/commits/b709ec32c710a846affe5f61ab70d726191ba673)

kwin commented on pom.xml:
```diff
@@ -23,7 +23,7 @@
         <groupId>org.apache.sling</groupId>
         <artifactId>sling-bundle-parent</artifactId>
-        <version>41</version>
+        <version>52</version>
```
Please also add property `project.build.outputTimestamp` for reproducible builds (https://maven.apache.org/guides/mini/guide-reproducible-builds.html#how-do-i-configure-my-maven-build).

rombert replied:
Updated in [1bfd1cf](https://github.com/apache/sling-org-apache-sling-serviceusermapper/pull/9/commits/1bfd1cfbfaa4869419d5b0d98212d6b7a3908f09)

rombert committed:
SLING-12004 - ServiceUserMapper fails to build with Java 17

- switch to a recent version of mockito
- update to a recent enough version of the parent pom
